Implement king square highlights for end game states

Bayuo321 opened the pull request:
- Implement findKingSquare in util.ts to locate king position from FEN
- Ensure chessground reloads after game ends (ctrl.ts)
- Differentiate highlights based on mate, timeout, resign or draw/stalemate (ground.ts)
- Created new endgame state icons (public/images/icons)
- Add SCSS pulsing animation with the SVG icons on king squares for enhanced visuals (_chessground.scss)

Co-authored-by: Pedro Antunes pedro.a.antunes@tecnico.ulisboa.pt
@Piti06

Bayuo321 committed:
Implement king square highlights for end game states

- Implement findKingSquare in util.ts to locate king position from FEN
- Ensure chessground reloads after game ends (ctrl.ts)
- Differentiate highlights based on mate, timeout, resign or
  draw/stalemate (ground.ts)
- Created new endgame state icons (public/images/icons)
- Add SCSS pulsing animation with the SVG icons on king squares for
  enhanced visuals (_chessground.scss)

Co-authored-by: Pedro Antunes <pedro.a.antunes@tecnico.ulisboa.pt>

kraktus commented:
Hey, thanks for the PR! I agree with the intention, but I don't think the flashing animation is the proper way to implement it, too distracting.

I'd rather have the information put like good/bad moves, etc
<img width="633" alt="image" src="https://github.com/user-attachments/assets/bc7ffae0-4f7f-4f47-89a2-168d6f44c6cd" />

It'd also reduce maintenance cost.

kraktus commented:
> Sorry if we missunderstood what you meant but our proposal is only related to when the game finishes.

I meant that the information should appear via a small information bubble like for good/bad moves. Not via a flashing square.

Piti06 commented:
Okay, we will work on that.

Bayuo321 committed:
fix: refine king square highlights for endgame states

- Updated endgame state icons
- Replaced infinite pulsing animation with a subtle one-time pop effect
- Added drop-shadow to king icons for improved visual emphasis

Co-authored-by: Pedro Antunes <pedro.a.antunes@tecnico.ulisboa.pt>

SergioGlorias commented:
@Bayuo321 It seems that there are some statuses missing that can happen, but they are without an icon.

Like VariantEnd and Cheat (It might be interesting to have a fallback in case it happens and not a specific one)

Can find status code here:
https://github.com/lichess-org/scalachess/blob/master/core/src/main/scala/Status.scala#L15-L27

SergioGlorias commented on public/images/icons/stalemate.svg:
This SVG is simply a PNG

Bayuo321 replied:
Hey!
I've changed the stalemate icon from ½ to handshake for clearer draw representation.
I think the handshake is a better fit.
This new icon, like the others, is an SVG from https://www.svgrepo.com and it's under the PD license.

Bayuo321 committed:
feat: update stalemate icon from ½ to handshake

Bayuo321 committed:
fix: add icons only for standard game variation

fitztrev commented:
Thanks for the PR. It seems like there might be support for it, just not implemented this way where it adds this amount of code and logic to maintain.
